File: lab02_stack/lab02_task03/scripts/subscriber_node.py
#!/usr/bin/env python
import rospy
##TODO: include the library for a msg typed VectorStamped

##TODO: complete your callback function. Your callback function should add the noisy numbers to the incoming x and z positions and print.
def callback(msg, args):
    print 'Remove this line'

def listener():

    rospy.init_node('listener', anonymous=True)

    ##TODO: Create two random numbers between 0 and 1.
    ##TODO: Create a subscriber and links to the callback function.

    rate = rospy.Rate(100000)
    ##TODO: Create a subscriber and links to the callback function.

    rospy.spin()
    rate.sleep()

if __name__ == '__main__':
    listener()

File: lab02/lab02task03/src/subscriber_node.cpp
#include "ros/ros.h"
//TODO: include the library for a msg typed VectorStamped

double x_noisy_pose = 0;
double z_noisy_pose = 0;

//TODO: complete your callback function. Your callback function should add the noisy numbers to the incoming x and z positions.
void subscribeCallback();

int main(int argc, char **argv)
{
    ros::init(argc, argv, "listener");

    ros::NodeHandle nh;

    //TODO: Create two random numbers between 0 and 1.

    //TODO: Create a subscriber and links to the callback function.
    while (nh.ok())
    {
        printf("The current point is (%.3f, %.3f)\n", x_noisy_pose, z_noisy_pose);
        ros::spinOnce();
        //Sleep for 20usec to prevent flooding
        usleep(20);
    }

    return 0;
}